holoviews/plotting/bokeh/element.py

```diff
@@ -618,6 +638,138 @@ def framewise(self):
+                                    'opts': {'location': (0, 0),
+                                             'orientation':'horizontal'}},
+                      'top':       {'pos': 'above',
+                                    'opts': {'location':(0, 0),
+                                             'orientation':'horizontal'}},
+                      'top_right':   {'pos': 'center',
+                                      'opts': {'location': 'top_right'}},
+                      'top_left':    {'pos': 'center',
+                                      'opts': {'location': 'top_left'}},
+                      'bottom_left': {'pos': 'center',
+                                      'opts': {'location': 'bottom_left',
+                                               'orientation': 'horizontal'}},
+                      'bottom_right': {'pos': 'center',
+                                      'opts': {'location': 'bottom_right',
+                                               'orientation': 'horizontal'}}}
+
+    colorbar = param.Boolean(default=False, doc="""
+        Whether to display a colorbar.""")
+
+    colorbar_position = param.ObjectSelector(objects=list(colorbar_specs),
+                                             default="right", doc="""
+        Allows selecting between a number of predefined colorbar position
+        options. The predefined options may be customized in the
+        colorbar_specs class attribute.""")
+
+    colorbar_opts = param.Dict(default={}, doc="""
+        Allows setting specific styling options for the colorbar overriding
+        the options defined in the colorbar_specs class attribute. Includes
+        location, orientation, height, width, scale_alpha, title, title_props,
+        margin, padding, background_fill_color and more.""")
+
+    logz  = param.Boolean(default=False, doc="""
+         Whether to apply log scaling to the z-axis.""")
+
+    _update_handles = ['color_mapper', 'source', 'glyph']
+
+    def _draw_colorbar(self, plot, color_mapper):
+        if LogColorMapper and isinstance(color_mapper, LogColorMapper):
+            ticker = LogTicker()
+        else:
+            ticker = BasicTicker()
+        cbar_opts = dict(self.colorbar_specs[self.colorbar_position],
+                         bar_line_color='black', label_standoff=8,
+                         major_tick_line_color='black')
+
+        # Check if there is a colorbar in the same position
+        pos = cbar_opts['pos']
+        if any(isinstance(model, ColorBar) for model in getattr(plot, pos, [])):
+            return
+
+        color_bar = ColorBar(color_mapper=color_mapper, ticker=ticker,
+                             **dict(cbar_opts['opts'], **self.colorbar_opts))
+
+        plot.add_layout(color_bar, pos)
+        self.handles['colorbar'] = color_bar
+
+
+    def _get_colormapper(self, dim, element, ranges, style):
+        low, high = ranges.get(dim.name)
+        palette = mplcmap_to_palette(style.pop('cmap', 'viridis'))
```

Comment:
It is easy to provide a Hot for bokeh:

Hot = ["black", "maroon", "darkred", "red", "orangered", "darkorange", "orange", "gold", "yellow", "white"]

I actually prefer this definition to the one in mpl, because it's more perceptually uniform; the one in mpl makes all the top few values be indistinguishably white on the monitors I have tested.  But it is slightly different than mpl's hot.

We could take mpl's hot and make that available for Bokeh easily by just copying the values from matplotlib/_cm_listed.py and converting them to a palette using `[(int(r*256),int(g*256),int(b*256)) for r,g,b in cm_data]`.  The advantage of doing this is that "hot" will mean the same thing in mpl as in HoloViews, and there's no existing "hot" in Bokeh to conflict.  So this way everyone would agree about the meaning of "hot".

But we can also have our own hot as listed above, perhaps renamed, to use as our default on both backends, with the advantage of being more perceptually uniform than mpl's hot, and being our own thing that won't have to be confused (at least not by name) with mpl's.  But because it's only *more* perceptually uniform, and isn't truly uniform, I'm not sure if that's the right thing to do.